packages/validator/src/services/prepareBeaconProposer.ts

```diff
@@ -1,47 +1,71 @@
-import {Epoch} from "@chainsafe/lodestar-types";
+import {Epoch, bellatrix} from "@chainsafe/lodestar-types";
 import {Api, routes} from "@chainsafe/lodestar-api";
-
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {fromHexString} from "@chainsafe/ssz";
+import {SLOTS_PER_EPOCH} from "@chainsafe/lodestar-params";
 import {IClock, ILoggerVc} from "../util/index.js";
 import {Metrics} from "../metrics.js";
 import {ValidatorStore} from "./validatorStore.js";
 import {IndicesService} from "./indices.js";
 
+const REGISTRATION_CHUNK_SIZE = 512;
 /**
  * This service is responsible for updating the BNs and/or Mev relays with
  * the corresponding feeRecipient suggestion. This should ideally run per epoch
  * but can be run per slot. Lighthouse also uses this to trigger any block
  */
 export class PrepareBeaconProposerService {
   constructor(
+    private readonly config: IBeaconConfig,
     private readonly logger: ILoggerVc,
     private readonly api: Api,
     private clock: IClock,
     private readonly validatorStore: ValidatorStore,
     private readonly indicesService: IndicesService,
-    private readonly metrics: Metrics | null
+    private readonly metrics: Metrics | null,
+    builderOpts: {enabled?: boolean}
   ) {
+    const currentEpoch = clock.getCurrentEpoch();
+
     clock.runEveryEpoch(this.prepareBeaconProposer);
+    void this.prepareBeaconProposer(currentEpoch);
+
+    if (builderOpts?.enabled) {
+      clock.runEveryEpoch(this.registerValidator);
+      void this.registerValidator(currentEpoch);
```

Comment:
Explain that `runEveryEpoch(this.registerValidator)` may take a while to run and requires to be run immediately because?

Reply:
because the validators could have a turn proposing before next slot's registration

Reply:
Not to me, but in a comment to the source code :pray: